src/Abstractions/NexusMods.Abstractions.MnemonicDB.Attributes/EnumAttribute.cs

```diff
@@ -0,0 +1,23 @@
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.ElementComparers;
+
+namespace NexusMods.Abstractions.MnemonicDB.Attributes;
+
+/// <summary>
+///    An attribute that represents an enum value.
+/// </summary>
+public class EnumAttribute<T>(string ns, string name) : ScalarAttribute<T, string>(ValueTags.Utf8, ns, name)
+    where T : Enum
+{
+    /// <inheritdoc />
+    protected override string ToLowLevel(T value)
+    {
+        return value.ToString();
+    }
+
+    /// <inheritdoc />
+    protected override T FromLowLevel(string value, ValueTags tag)
+    {
+        return (T) Enum.Parse(typeof(T), value);
+    }
```

Comment:
Converting Enums to and from strings is notoriously slow, with tons of allocations. Using strings also means that we'll never be able to change the name in the enum or we'll break this. Instead of using names, it'd be better to use the underlying integer values. The integer values are more resilient to change because developers are already accustomed to not changing existing values.

@Sewer56 probably has some more details on this.